changelogs/unreleased/uri-library-extension.md

```diff
@@ -0,0 +1,12 @@
+## feature/uri
```

Comment:
I tried this:
```Lua
uri.parse_many({
    "/tmp/unix.sock?q1=v1",
    params = {q2 = 1LL}
})
```
and I didn't get an error neither a URI object. Just null. Firstly, the params were valid. I just used a cdata integer instead of a Lua number. Secondly, null is not an option in both our guidelines - it must be either a raised exception or `nil,err`. I have no idea which one is the currently selected. Probably the exceptions.

Reply:
1. About cdata i didn't think it was worth supporting, if you think I'll add this feature.
2. I specially return 0 without any error, because `uri.parse` do it! I wan't to have same behaviour for this functions and uri.parse doesn't raised exception or return error. If you are not afraid of breaking backward compatibility, I would fix it everywhere

Reply:
> Firstly, the params were valid. I just used a cdata integer instead of a Lua number.

AFAICS we usually don't handle int64 unless we specifically need to:

```
tarantool> require('fiber').cond():wait(1)
---
- false
...

tarantool> require('fiber').cond():wait(1LL)
---
- error: 'usage: cond:wait([timeout])'
...
```

I'd say that a parameter value must always be a string - after all `uri_param::value` is a string in C. Number is implicitly convertible to string by `lua_tostring` so I think it's okay to allow numbers as well, but not int64.

> I specially return 0 without any error, because `uri.parse` do it!

I think it should be fine to return an error as a second value - existing users wouldn't notice while new users could make use of it.

Reply:
Fixed, return error as a second parameter to prevent breaking backward compatibility